themes/src/main/resources/theme/base/login/resources/js/utils.js

```diff
@@ -0,0 +1,88 @@
+/*
+ *
+ *  * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ *  * and other contributors as indicated by the @author tags.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+const formatNumber = (input, format) => {
+    if (!input) {
+        return "";
+    }
+
+    // array holding the patterns for the number of expected digits in each part
+    const digitPattern = format.match(/{\d+}/g);
+
+    if (!digitPattern) {
+        return "";
+    }
+
+    let maxSize = 0;
+
+    // calculate the maximum size of the given pattern based on the sum of the expected digits
+    for (const p of digitPattern) {
+        maxSize += parseInt(p.replace("{", "").replace("}", ""));
+    }
+
+    // keep only digits
+    let rawValue = input.replace(/\D+/g, '');
+
+    // make sure the value is a number
+    if (parseInt(rawValue) != rawValue) {
```

Comment:
Perhaps the global [`isNaN()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#examples) could be used here as well?
```suggestion
    if (!isNaN(rawValue)) {
```

Reply:
Does not work well for me.

Reply:
Ok, let's leave this as-is then.